src/test/java/org/broadinstitute/dsm/ElasticSearchTest.java

```diff
@@ -740,6 +746,62 @@ public void updateWorkflowValues() throws Exception {
             }
         }
         Assert.assertEquals(workflows.size(), updatedWorkflows.size());
+
+        ElasticSearchUtil.updateRequest(ddpParticipantId, ddpInstance.getParticipantIndexES(), workflowsBefore);
+    }
+
+    @Test
+    public void updateWorkflowValuesWithStudySpecificData() throws Exception {
+        String ddpParticipantId = "XLDUNC3BHGWGWERHW781";
+        String workflow = "ALIVE_DECEASED";
+        String status = "ALIVE";
+        String subjectId = "testId";
+        String firstname = "testfirstname";
+        String lastname = "testlastname";
+        DDPInstance ddpInstance = new DDPInstance(null,null, null, null, false, 0, 0,
+                false, null, false, null, "participants_structured.rgp.rgp", null, null);
+        Map<String, Object> workflowsBefore = ElasticSearchUtil.getObjectsMap(ddpInstance.getParticipantIndexES(), ddpParticipantId, "workflows");
+
+        ElasticSearchUtil.writeWorkflow(WorkflowForES.createInstanceWithStudySpecificData(ddpInstance, ddpParticipantId,
+                workflow, status, new WorkflowForES.StudySpecificData(subjectId, firstname, lastname)));
+
+        testWorkflowWithStudySpecificData(ddpParticipantId, workflow, status, subjectId, firstname, lastname, ddpInstance);
+
+        String newSubjectId = "testId3";
+
+        ElasticSearchUtil.writeWorkflow(WorkflowForES.createInstanceWithStudySpecificData(ddpInstance, ddpParticipantId,
+                workflow, status, new WorkflowForES.StudySpecificData(newSubjectId, firstname, lastname)));
+
+        testWorkflowWithStudySpecificData(ddpParticipantId, workflow, status, subjectId, firstname, lastname, ddpInstance);
+
+        ElasticSearchUtil.updateRequest(ddpParticipantId, ddpInstance.getParticipantIndexES(), workflowsBefore);
+
+    }
+
+    public void testWorkflowWithStudySpecificData(String ddpParticipantId, String workflow, String status, String subjectId, String firstname, String lastname, DDPInstance ddpInstance) throws Exception {
+        Map<String, Object> workflows = ElasticSearchUtil.getObjectsMap(ddpInstance.getParticipantIndexES(), ddpParticipantId, "workflows");
+        Assert.assertTrue(workflows != null && !workflows.isEmpty());
+        if (workflows != null && !workflows.isEmpty()) {
```

Comment:
What's the point checking `workflows != null && !workflows.isEmpty()` here when one line above you already check it?